src/unity/toolkits/neural_net/CMakeLists.txt

```diff
@@ -7,68 +7,35 @@ project(unity_toolkits)
 
 # When building for macOS, build the MetalPerformanceShaders-based backend.
 if(APPLE AND HAS_MPS AND NOT TC_BUILD_IOS)
-    find_library(ACCELERATE NAMES Accelerate)
-    message("Accelerate found at ${ACCELERATE}.")
 
-    find_library(CORE_GRAPHICS NAMES CoreGraphics)
-    message("CoreGraphics found at ${CORE_GRAPHICS}.")
-
-    find_library(CORE_IMAGE NAMES CoreImage)
-    message("CoreImage found at ${CORE_IMAGE}.")
-
-    find_library(METAL NAMES Metal)
-    message("Metal found at ${METAL}.")
-
-    find_library(METAL_PERFORMANCE_SHADERS NAMES MetalPerformanceShaders)
-    message("MetalPerformanceShaders found at ${METAL_PERFORMANCE_SHADERS}.")
-
-    make_library(tcmps
-        SOURCES
-        TCMPSImageAugmenting.m
-        mps_image_augmentation.mm
-        mps_compute_context.mm
-        mps_graph_trainer.mm
-        mps_graph_networks.mm
-        mps_updater.mm
-        mps_utils.mm
-        mps_graph_cnnmodule.mm
-        mps_layers.mm
-        mps_networks.mm
-        mps_trainer.mm
-        mps_lstm_helper.mm
-        mps_cnnmodule.mm
-        mps_graph_layers.mm
-        mps_weight.mm
-        mps_device_manager.m
-      REQUIRES
-        unity_core
-        ${ACCELERATE}
-        ${CORE_GRAPHICS}
-        ${CORE_IMAGE}
-        ${METAL}
-        ${METAL_PERFORMANCE_SHADERS}
+    add_library(tcmps OBJECT
+      TCMPSImageAugmenting.m
+      mps_image_augmentation.mm
+      mps_compute_context.mm
+      mps_graph_trainer.mm
+      mps_graph_networks.mm
+      mps_updater.mm
+      mps_utils.mm
+      mps_graph_cnnmodule.mm
+      mps_layers.mm
+      mps_networks.mm
+      mps_trainer.mm
+      mps_lstm_helper.mm
+      mps_cnnmodule.mm
+      mps_graph_layers.mm
+      mps_weight.mm
+      mps_device_manager.m
     )
+    add_dependencies(tcmps boost)
     target_compile_options(tcmps PUBLIC "-fobjc-arc -Wno-nullability-completeness -Wno-shadow-ivar")
-    set(additional_unity_neural_net_requirements tcmps)
-else()
-    set(additional_unity_neural_net_requirements "")
 endif()
 
 # TODO: a backend wrapping MXNet.
 
-make_library(unity_neural_net
-  SOURCES
-    compute_context.cpp
-    float_array.cpp
-    image_augmentation.cpp
-    model_spec.cpp
-  REQUIRES
-    image_type
-    logger
-    unity_core
-    unity_coreml_model_export
-    ${additional_unity_neural_net_requirements}
-  COMPILE_FLAGS_EXTRA_GCC
-    -Wno-unknown-pragmas  # NOTE: used for auto-generated protobuf source files
-    -Wno-unused-function  # NOTE: used for auto-generated protobuf source files
+add_library(unity_neural_net OBJECT
+  compute_context.cpp
+  float_array.cpp
+  image_augmentation.cpp
+  model_spec.cpp
 )
+add_dependencies(unity_neural_net boost)
```

Comment:
Why do we need to add this to each object library?

Reply:
To enforce ordering. Otherwise the boost headers don’t exist yet (in the place we’re looking for them). It’s possible we could refer to them elsewhere but I haven’t tried.

Reply:
Makes sense. Room for future work